python/update_package_properties.py

```diff
@@ -0,0 +1,301 @@
+    changelog = f"{latest_changelog}\n\n{rpm_changelog_history(spec_full_path)}"
+    content = template.render(version=package_properties_params.project_version, rpm_version=rpm_version,
+                              project_name=package_properties_params.project_name,
+                              fancy_version_no=package_properties_params.fancy_version_number, changelog=changelog)
+    with open(spec_full_path, "w+") as writer:
+        writer.write(content)
+
+
+@validate_parameters
+def update_all_changes(github_token: non_empty(non_blank(str)), project_name: non_empty(str),
+                       project_version: is_version(str),
+                       tag_name: is_tag(str), fancy: bool, fancy_version_number: non_negative(int),
+                       microsoft_email: is_email(str),
+                       name_surname: non_empty(non_blank(str)), release_date: datetime, packaging_path: str):
+    templates_path = f"{BASE_PATH}/templates"
+    update_pkgvars(project_name, project_version, fancy, fancy_version_number, templates_path, f"{packaging_path}")
+    latest_changelog = changelog_for_tag(github_token, project_name, tag_name)
+    package_properties_params = PackagePropertiesParams(project_name=project_name, project_version=project_version,
+                                                        fancy=fancy, fancy_version_number=fancy_version_number,
+                                                        microsoft_email=microsoft_email, name_surname=name_surname,
+                                                        changelog_date=release_date, latest_changelog=latest_changelog)
+
+    prepend_latest_changelog_into_debian_changelog(package_properties_params, f"{packaging_path}/debian/changelog")
+    spec_full_path = f"{packaging_path}/{package_properties_params.spec_file_name()}"
+    update_rpm_spec(package_properties_params, spec_full_path, templates_path)
+
+
+# update_all_changes("93647419346e194ae3094598139d68ebf2263ee0", "citus", "10.0.3", "v10.0.3", True, 1,
+#                    "gindibay@microsoft.com", "Gurkan Indibay")
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--gh_token')
+    parser.add_argument('--prj_name')
+    parser.add_argument('--prj_ver')
+    parser.add_argument('--tag_name')
+    parser.add_argument('--fancy')
+    parser.add_argument('--fancy_ver_no')
+    parser.add_argument('--email')
+    parser.add_argument('--name')
+    parser.add_argument('--date', nargs='?', default=datetime.now())
+    parser.add_argument('--exec_path')
+    arguments = parser.parse_args()
+
+    if not string_utils.is_integer(arguments.fancy_ver_no):
+        raise ValueError(f"fancy_ver_no is expected to be numeric actual value {arguments.fancy_ver_no}")
+
+    exec_date = datetime.strptime(arguments.date, '%Y.%m.%d %H:%M:%S %z')
+
+    update_all_changes(arguments.gh_token, arguments.prj_name, arguments.prj_ver, arguments.tag_name, arguments.fancy,
```

Comment:
It would be nicer if we passed the `arguments` object as a parameter for readability purposes here.
It is quite possible to break the order of parameters here.

`update_all_changes(arguments)` would be more error prune I think

Reply:
Fixed